### PR TITLE
ccache-4.8.3 new package & hiredis-1.2.0 new package

### LINKED ISSUE #6340: [Wolfi Package Request]: $ccache
### What software would you like us to add to wolfi-os. Ideally include a URL to the project and its source.

https://ccache.dev/

Ccache is a compiler cache. It [speeds up recompilation](https://ccache.dev/performance.html) by caching previous compilations and detecting when the same compilation is being done again.

### which versions of the software should we include?

4.8.3

### Add some justification for why this specific package and versions are important.

This is the latest version. 

### Please check all that apply

- [X] This package has an un-restrictive license
- [X] The package/versions proposed are actively maintained upstream
- [X] I am interested in adding this package to Wolfi OS myself
- [X] I am willing to help maintain this package

### PR DESCRIPTION
**Pull request by debasishbsws**
Fixes: #6340 

#### For new package PRs only

- [x] This PR is marked as fixing a pre-existing package request bug
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [x] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

Also added `hiredis` a minimalistic C client library for Redis.  
https://redis.com/lp/hiredis/

#### For new package PRs only

- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates

### CODE REVIEW

File: ccache.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/lib/ccache/bin
+      for link in cc gcc g++ cpp c++ clang clang++ c89 c99; do
+          ln -sf ${{targets.destdir}}/bin/ccache ${{targets.destdir}}/usr/lib/ccache/bin/${link}
+      done
+
+update:
+  enabled: true
+  github:
+    identifier: ccache/ccache
```
Comment (ajayk) [inline ccache.yaml:48]:
```suggestion
update:
  enabled: true
  github:
    identifier: ccache/ccache
    strip-prefix: v
    use-tag: true
```

File: hiredis.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/redis/hiredis
+      tag: v${{package.version}}
+      expected-commit: 60e5075d4ac77424809f855ba3e398df7aacefe8
+
+  - runs: |
+      # make USE_SSL=1 PREFIX=/usr DEBUG="$CFLAGS" LDFLAGS="$LDFLAGS"
+
```
Comment (ajayk) [inline hiredis.yaml:26]:
```suggestion

```

File: hiredis.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+subpackages:
+  - name: hiredis-dev
+    description: "hiredis dev headers"
+    pipeline:
+      - uses: split/dev
+
+update:
+  enabled: true
+  github:
+    identifier: redis/hiredis
```
Comment (ajayk) [inline hiredis.yaml:38]:
```suggestion
  github:
    identifier: redis/hiredis
    strip-prefix: v
    use-tag: true
```

Comment (debasishbsws) [reply hiredis.yaml:38]: hey @ajayk where do I learn more about this update section. There are many things I am not sure what they do like `strip-prefix`, `tag-filter` etc. [Chainguard Academy](https://edu.chainguard.dev/open-source/wolfi/building-a-wolfi-package/#the-update-section) doesn't describe much.

**Review by ajayk — APPROVED** [review]

**Pull request merged**